src/charm.py

```diff
@@ -930,6 +974,20 @@ def _generate_middleware_config(
           "cannot create middleware: multi-types middleware not supported, consider declaring two
           different pieces of middleware instead"
         """
+        forwardauth_middleware = {}
+        if self.forward_auth.is_ready():
+            forward_auth_config = self.forward_auth.get_provider_info()
+            if data.get("name") in forward_auth_config.app_names:  # type: ignore
+                policy_decision_point_app = self.forward_auth.get_remote_app_name()
```

Comment:
`app_name` could be confused with `forward_auth_config.app_names`. I think the original variable name is more descriptive because it points to oathkeeper as the policy decision point (which is what forwardAuth talks to in IAP architecture, and thus we want to include it in the middleware name).